udata_front/theme/gouvfr/assets/js/components/AdminTable/AdminDiscussionsTable/AdminDiscussionsTable.vue

```diff
@@ -0,0 +1,159 @@
+    <tbody>
+      <tr v-for="discussion in discussions">
+        <td>
+          <p class="fr-text--bold">
+            <TextClamp
+              :text="discussion.title"
+              :auto-resize="true"
+              :max-lines="1"
+            />
+          </p>
+          <p v-if="subjects[discussion.subject.id]">
+            <a
+              class="fr-link inline-flex"
+              :href="subjects[discussion.subject.id].page"
+            >
+              <Vicon :height="12" class="self-center" :name="subjects[discussion.subject.id].icon"/>
+              <TextClamp
+                class="overflow-wrap-anywhere"
+                :text="subjects[discussion.subject.id].title"
+                :auto-resize="true"
+                :max-lines="1"
+              />
+            </a>
+          </p>
+        </td>
+        <td>
+          <AdminBadge :type="getStatus(discussion).type">
+            {{ getStatus(discussion).label }}
+          </AdminBadge>
+        </td>
+        <td>{{ formatDate(discussion.created) }}</td>
+        <td>
+          <template v-if="discussion.closed">{{ formatDate(discussion.closed) }}</template></td>
+        <td>
+          <a
+            v-if="subjects[discussion.subject.id]"
+            class="fr-link inline-flex"
+            :href="getDiscussionUrl(discussion.id, subjects[discussion.subject.id])"
+            target="_blank"
+          >
+            <TextClamp
+              :text="getDiscussionUrl(discussion.id, subjects[discussion.subject.id])"
+              :auto-resize="true"
+              :max-lines="1"
+            />
+          </a>
+        </td>
+      </tr>
+    </tbody>
+  </AdminTable>
+</template>
+<script setup lang="ts">
+import { formatDate } from "@datagouv/components/ts";
+import { OhVueIcon as Vicon } from "oh-vue-icons";
+import { ref, watchEffect } from "vue";
+import { useI18n } from "vue-i18n";
+import TextClamp from "vue3-text-clamp";
+import { formatSubject, getSubject, SubjectSummary } from "../../../api/discussions";
+import AdminTable from "../Table/AdminTable.vue";
+import AdminTableTh from "../Table/AdminTableTh.vue";
+import type { AdminBadgeState, DiscussionSortedBy, DiscussionSubjectTypes, SortDirection, Thread } from '../../../types';
+import { user } from "../../../config";
+import AdminBadge from "../../AdminBadge/AdminBadge.vue";
+
+const props = defineProps<{
+  discussions: Array<Thread>;
+  loading: boolean;
+  sortDirection: SortDirection;
+  sortedBy: DiscussionSortedBy;
+}>();
+
+defineEmits<{
+  (event: 'sort', column: DiscussionSortedBy, direction: SortDirection): void;
+}>();
+
+const { t } = useI18n();
+
+const subjectPromises = ref<Record<string, Promise<DiscussionSubjectTypes>>>({});
+const subjects = ref<Record<string, SubjectSummary>>({});
+
+watchEffect(async () => {
+  for (const discussion of props.discussions) {
+    if(!(discussion.subject.id in subjectPromises.value)) {
+      const subjectPromise = getSubject(discussion.subject);
```

Comment:
If the promise was sent but is still waiting for a response, we can reuse the same promise to have the result for other discussions on the same subject without additional request.

We can't check for the id in `subjects.value` because this one is filled only when the response is received.